darshan-runtime/lib/darshan-ldms.c

```diff
@@ -0,0 +1,249 @@
+                dC.conn_status = 0;
+                break;
+        case LDMS_XPRT_EVENT_REJECTED:
+                ldms_xprt_put(x);
+                dC.conn_status = ECONNREFUSED;
+                break;
+        case LDMS_XPRT_EVENT_DISCONNECTED:
+                ldms_xprt_put(x);
+                dC.conn_status = ENOTCONN;
+                break;
+        case LDMS_XPRT_EVENT_ERROR:
+                dC.conn_status = ECONNREFUSED;
+                break;
+        case LDMS_XPRT_EVENT_RECV:
+                sem_post(&dC.recv_sem);
+                break;
+        case LDMS_XPRT_EVENT_SEND_COMPLETE:
+                break;
+        default:
+                printf("Received invalid event type %d\n", e->type);
+        }
+}
+
+#define SLURM_NOTIFY_TIMEOUT 5
+ldms_t setup_connection(const char *xprt, const char *host,
+                        const char *port, const char *auth)
+{
+        char hostname[PATH_MAX];
+        const char *timeout = "5";
+        int rc;
+        struct timespec ts;
+
+        if (!host) {
+                if (0 == gethostname(hostname, sizeof(hostname)))
+                        host = hostname;
+        }
+        if (!timeout) {
+                ts.tv_sec = time(NULL) + 5;
+                ts.tv_nsec = 0;
+        } else {
+                int to = atoi(timeout);
+                if (to <= 0)
+                        to = 5;
+                ts.tv_sec = time(NULL) + to;
+                ts.tv_nsec = 0;
+        }
+
+        ldms_g = ldms_xprt_new_with_auth(xprt, NULL, auth, NULL);
+        if (!ldms_g) {
+                printf("Error %d creating the '%s' transport\n",
+                       errno, xprt);
+                return NULL;
+        }
+
+        sem_init(&dC.recv_sem, 1, 0);
+        sem_init(&dC.conn_sem, 1, 0);
+
+        rc = ldms_xprt_connect_by_name(ldms_g, host, port, event_cb, NULL);
+        if (rc) {
+                printf("Error %d connecting to %s:%s\n",
+                       rc, host, port);
+                return NULL;
+        }
+        sem_timedwait(&dC.conn_sem, &ts);
+        if (dC.conn_status)
+                return NULL;
+        return ldms_g;
+}
+
+void darshan_ldms_connector_initialize()
+{
+    if (!getenv("DARSHAN_LDMS_STREAM"))
+        dC.env_ldms_stream = "darshanConnector";
+
+    /* Set flags for various LDMS environment variables */
+    if (getenv("DXT_ENABLE_LDMS"))
+        dC.dxt_enable_ldms = 0;
+    else
+        dC.dxt_enable_ldms =1;
+
+    if (getenv("POSIX_ENABLE_LDMS"))
+        dC.posix_enable_ldms = 0;
+    else
+        dC.posix_enable_ldms = 1;
+
+    if (getenv("MPIIO_ENABLE_LDMS"))
+        dC.mpiio_enable_ldms = 0;
+    else
+        dC.mpiio_enable_ldms = 1;
+
+    if (getenv("STDIO_ENABLE_LDMS"))
+        dC.stdio_enable_ldms = 0;
+    else
+        dC.stdio_enable_ldms = 1;
+
+    if (getenv("HDF5_ENABLE_LDMS"))
+        dC.hdf5_enable_ldms = 0;
+    else
+        dC.hdf5_enable_ldms = 1;
+
+    if (getenv("MDHIM_ENABLE_LDMS"))
```

Comment:
I would actually recommend dropping the MDHIM support, too. This isn't a module that is used for much of anything at this point, so let's not even worry about maintaining LDMS support there. The code won't ever get used, so no point in maintaining it.